universal/src/logging/json_string.cpp

```diff
@@ -0,0 +1,50 @@
+#include "userver/logging/json_string.hpp"
+
+#include <userver/logging/log.hpp>
+#include <userver/utils/assert.hpp>
+
+#include <algorithm>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace logging {
+
+namespace {
+constexpr std::string_view kNull = "null";
+
+bool DebugValid(std::string_view json) {
+  try {
+    formats::json::FromString(json);
+  } catch (const formats::json::ParseException& error) {
+    LOG_ERROR() << "Invalid json: " << json << ", error: " << error;
```

Comment:
This log should somehow be moved into the assert message

One way to do that:

```cpp
bool IsValidJsonStringDebug(std::string_view json, std::string& what);

//...

std::string what;
UASSERT_MSG(IsValidJsonStringDebug(json_, what), fmt::format("Invalid json: {}, error: {}", json_, what));
```
```

Reply:
fixed